src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -58,6 +56,8 @@ interface IncomingPaymentsDb {
      */
     suspend fun receivePayment(paymentHash: ByteVector32, receivedWith: List<IncomingPayment.ReceivedWith>, receivedAt: Long = currentTimestampMillis())
 
+    suspend fun receiveOfferPayment(origin: IncomingPayment.Origin.Offer, preimage: ByteVector32, receivedWith: List<IncomingPayment.ReceivedWith>, receivedAt: Long = currentTimestampMillis())
```

Comment:
Actually I'm even wondering if that additional function is useful or even desirable. We rely on calling `receivePayment` multiple times in the pay-to-open case. You also rely on `getIncomingPayment` in the `IncomingPaymentHandler`, which means it would have to be connected to the tables used by `receivePayment` *and* `receiveOfferPayment`.

Why don't we instead:

- remove `receiveOfferPayment`
- in the `IncomingPaymentHandler`, when accepting an offer payment, call `addIncomingPayment` first and then `receivePayment`

This is the flow we use for payments received by on-chain swaps, it's more general, and this way we ensure there's no inconsistency during pay-to-open or other on-the-fly funding schemes. It's also easier to understand for wallets who need to implements the `PaymentsDb` trait. What do you think?